components/breadcrumbs/breadcrumb.js

```diff
@@ -0,0 +1,89 @@
+import '../icons/icon.js';
+import '../link/link.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+class BreadCrumb extends RtlMixin(LitElement) {
+
+	static get properties() {
+		return {
+			compact: {
+				type: Boolean,
+				reflect: true
+			},
+			href: {
+				type: String,
+				reflect: true
+			},
+			target: {
+				type: String,
+				reflect: true
+			},
+			text: {
+				type: String,
+				reflect: true
+			},
+			ariaLabel: {
+				attribute: 'aria-label',
+				type: String,
+				reflect: true
+			}
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				align-items: center;
+				display: inline-flex;
+			}
+
+			:host([compact]) .d2l-breadcrumb-wrapper {
+				display: flex;
+				flex-direction: row-reverse;
+				align-items: center;
+			}
+
+			d2l-icon {
+				height: 8px;
+				padding-left: 10px;
+				padding-right: 5px;
+				width: 8px;
+			}
+
+			:host([dir="rtl"]) d2l-icon {
+				padding-left: 5px;
+				padding-right: 10px;
+			}
+
+			:host([compact]) d2l-icon {
+				padding-right: 10px;
+				padding-left: 0;
+			}
+
+			:host([dir="rtl"][compact]) d2l-icon {
+				padding-right: 0;
+				padding-left: 10px;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.href = '#';
+		this.compact = false;
+		this.text = '';
+	}
+
+	render() {
+		return html`
+			<div class="d2l-breadcrumb-wrapper" role="navigation" aria-label="${this.ariaLabel ? this.ariaLabel : this.text}">
```

Comment:
I know this was just copied from the Polymer element, but this might be a good time to align the accessibility with [ARIA best practices](https://www.w3.org/TR/wai-aria-practices/examples/breadcrumb/index.html).

They recommend that the `navigation` role actually be the parent -- we could turn `<div class="d2l-breadcrumbs-wrapper">` into a `<nav>`. That navigation element is the one that needs a label, but it seems like they recommend just using "Breadcrumb" as a generic label always, so that lang term could just live inside this component.

They further recommend using an unordered list and markup up the last non-link breadcrumb with `aria-current="page"`. I don't _think_ these breadcrumbs have that concept?

All of this actually aligns with how we do breadcrumbs in the monolith. Here's the markup they generate:
```html
<nav aria-labelledby="d2l_1_31_683">
  <h2 class="d2l-heading vui-offscreen" id="d2l_1_31_683">Breadcrumb:</h2>
  <ol aria-labelledby="d2l_1_31_683">
    <li><a class="d2l-link" href="/d2l/link1">Link 1</a></li>
    <li><a class="d2l-link" href="/d2l/link2">Link 2</a></li>
    <li><span class="d2l-breadcrumb-text">Page</span></li>
  </ol>
</nav>
```

We can also proceed as-is and Gaudi can look into making these ARIA changes afterwards.

Reply:
I think I managed to make the parent to be `<nav>` and the slot to be rendered inside and `<ol>`
so this would be the parent:
```
<nav aria-label="Breadcrumb" class="d2l-breadcrumbs-wrapper" @d2l-breadcrumb-connected="${this._handleBreadCrumbConnected}">
				<ol>
				<slot></slot>
				</ol>
			</nav>
```
these extra styles added to parent as well:
```
.d2l-breadcrumbs-wrapper ol {
				margin: 0;
				padding-left: 0;
				list-style: none;
			}
```
and then the child would look like this:
```
<li class="d2l-breadcrumb-wrapper" ?data-compact=${this._compact} role="navigation">
				<d2l-link aria-label="${this.ariaLabel ? this.ariaLabel : this.text}" href="${this.href}" target="${ifDefined(this.target)}">${this.text}</d2l-link>
				${this._compact ? html`<d2l-icon icon="d2l-tier1:chevron-left"></d2l-icon>` : html`<d2l-icon icon="d2l-tier1:chevron-right"></d2l-icon>`}
			</li>
```
but it doesn't cover the fact that we should set `aria-current="page"` unless the user of the component sets it properly. Thoughts?

Reply:
Also the separators are added with css in the original suggestion but we have them as icons. Do we want to change that?

Reply:
Cool! Don't worry about the `aria-current="page"` for now. Based on what [design has specified](http://design.d2l/components/breadcrumbs/), they don't want it:
> Do not include the current page title in the breadcrumb as a non-clickable link – repetition is repetitive

For the list part, I'd want to test it out in a screen reader, but I think having `<ol><d2l-breadcrumb><li>` in the resolved DOM is going to trip them up -- list items are supposed to be direct children of the list itself, which doesn't play nice across the shadow boundaries. There are a couple ways I think we could solve that:
1. `<ol><d2l-breadcrumb role="presentation"><li>` will tell the screen reader to ignore that element in terms of semantics.
2. `<div role="list"><d2l-breadcrumb role="listitem">` applies the list/listitem roles to existing elements

In both cases, our `d2l-breadcrumb` element is "sprouting" an attribute, which isn't amazing but I don't really see an alternative. @dbatiste not sure if you have thoughts there.

Haha and we thought breadcrumbs would be an easy conversion... now you're learning all about accessibility and ARIA! 🎉

Reply:
No other options to offer up here, other than dropping the list rendering altogether, which does not align with Aria example.  In either case above it would require sprouting, so I would be inclined to sprout `role="listitem"`, but perhaps still try to use the `ol`.  While not ideal, we've had to sprout role in a few other places.  The consumer could mess with it, but if they do (unlikely), they are doing bad things.

Reply:
I went with the first option (setting a presentation role for breadcrumb in the middle per Dave's suggestion)

Reply:
Is that preferred?  How does it announce screenreader-wise?

Reply:
@dbatiste I tested with chromevox and it doesn't appear to read out the breadcrumbs part. I guess it doesn't work as expected?

Reply:
I tried the other way as well. The screen reader doesn't seem to read it this way either:

```
			<div class="d2l-breadcrumbs-wrapper" role="list" aria-label="Breadcrumbs" @d2l-breadcrumb-connected="${this._handleBreadCrumbConnected}">
				<slot></slot>
			</div>
```

```
.d2l-breadcrumbs-wrapper {
				list-style: none;
				margin: 0;
				padding: 0;
				overflow: hidden;
				position: relative;
				white-space: nowrap;
			}
```

```
<div class="d2l-breadcrumb-wrapper" ?data-compact=${this._compact} role="navigation">
				<d2l-link aria-label="${this.ariaLabel ? this.ariaLabel : this.text}" href="${this.href}" target="${ifDefined(this.target)}">${this.text}</d2l-link>
				${this._compact ? html`<d2l-icon icon="d2l-tier1:chevron-left"></d2l-icon>` : html`<d2l-icon icon="d2l-tier1:chevron-right"></d2l-icon>`}
			</div>
```

Reply:
The child `div` there needs `role="listitem"` not `navigation`.

Reply:
So Dave took care of it himself . So ✔️